javascript_es6/1.a.browser-echobot/src/webChatAdapter.js

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ConnectionStatus } from 'botframework-webchat';
+import { Subject, Observable, BehaviorSubject } from 'rxjs';
+import { BotAdapter, TurnContext } from 'botbuilder-core';
+
+/**
+ * Custom BotAdapter used for deploying a bot in a browser.
+ */
+export class WebChatAdapter extends BotAdapter {
+    constructor() {
+        super();
+        this.activity$ = new Subject();
+        this.botConnection = {
+            connectionStatus$: new BehaviorSubject(ConnectionStatus.Online),
+            activity$: this.activity$.share(),
+            end() {
+                debugger
+            },
+            postActivity: activity => {
+                const id = Date.now().toString();
+                return Observable.fromPromise(this
+                    .onReceive(Object.assign({}, activity, {
+                        id,
+                        conversation: { id: 'bot' },
+                        channelId: 'WebChat'
+                    }))
+                    .then(() => id)
+                )
+            }
+        }
+    }
+
+    /**
+     * This WebChatAdapter implements the sendActivities method which is called by the TurnContext class.
+     * It's also possible to write a custom Context class with different methods of accessing an adapter.
+     * @param {TurnContext} context
+     * @param {Activity[]} activities
+     */
+    sendActivities(context, activities) {
+        console.log(Date.now().toString());
+        const sentActivities = activities.map(activity => Object.assign({}, activity, {
+            id: Date.now().toString(),
+            channelId: 'WebChat',
+            conversation: { id: 'bot' },
+            from: { id: 'bot' },
+            timestamp: Date.now()
+        }));
+
+        sentActivities.forEach(activity => this.activity$.next(activity));
+
+        // Returns a mock of the ResourceResponse (https://github.com/Microsoft/botbuilder-js/blob/master/libraries/botframework-schema/src/index.ts#L605-L616)
```

Comment:
>Returns a mock of the R [](start = 11, length = 23)

Why would we want a mock of the Resource Response? (real question)

Reply:
I need to talk to someone about what this is in the regular REST API context. The only description we have for a [ResourceResponse is that it can be used to get an attachment that has been uploaded](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-api-reference?view=azure-bot-service-3.0#upload-attachment-to-channel). 

I'm inferring that it's an ActivityId in general (and this in this case) that can be used to update and delete activities.